frontend/src/components/Collections/components/Grid/common/utils.ts

```diff
@@ -34,6 +35,12 @@ export function aggregateDatasetsMetadata(datasets: Dataset[]) {
     organism: unique(result.organism),
     tissue: unique(result.tissue),
   };
+}, hashFn);
+
+function hashFn(datasets: Dataset[]): string {
+  return datasets.reduce((acc, dataset) => {
+    return acc + dataset.id;
+  }, "");
```

Comment:
hash function is just a string of all ids, this implementation is sensitive to ordering, but that shouldn't be changing between api calls